ocp_utilities/infra.py

```diff
@@ -49,6 +53,30 @@ def assert_nodes_schedulable(nodes):
         )
 
 
+def assert_pods_failed_or_pending(pods: list) -> None:
+    """
+    Validates all pods and flag failed or pending pods
+
+    Args:
+         pods: List of pod objects
+
+    Raises:
+        PodsFailedOrPendingError, if there are failed or pending pods
+    """
+    LOGGER.info("Verify all pods are not failed nor pending.")
+
+    failed_or_pending_pods = []
+    for pod in pods:
```

Comment:
please use list comprehension to save all failed/pending pods

Reply:
if I include the namespace as optional, then I propose:
```suggestion
    failed_or_pending_pods = []
    for pod in pods:
        if namespace and pod.namespace != namespace:
            continue
        if pod.exists and pod.instance.status.phase in [pod.Status.PENDING, pod.Status.FAILED]:
            failed_or_pending_pods.append(f"{pod.name} is {pod.instance.status.phase}")
```